src/Interactive/Host/Microsoft.CodeAnalysis.InteractiveHost.csproj

```diff
@@ -37,6 +37,7 @@
     <Reference Include="System.Windows.Forms" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="..\..\Compilers\Core\Portable\InternalUtilities\NonCopyableAttribute.cs" Link="Utilities\NonCopyableAttribute.cs" />
```

Comment:
I don't think injecting this attribute explicitly into every project that needs this analysis is a good approach. Can we follow the same model as PerformanceSensitiveAttribute? We should unify our approach for similar kinds of custom attribute based analyses.
https://github.com/dotnet/roslyn/blob/96af121422a90b0c63b1c675005ea9ddb4d6a2d3/eng/targets/Settings.props#L140-L146

Reply:
This is what we already do for `[NonDefaultable]`. We can leave this for a follow-up for now.

Reply:
I don't like building up this debt, but I'll let you and @genlu make a call here. Analyzers requiring such custom annotation attributes for analysis are only going to increase with time, so this approach of analyzer consumers having to define the attribute and include them in required project files does not look scalable.

Reply:
I agree with Manish, we should follow the same pattern from PerfSensitiveAnalyzer before releasing the analyzer.

Reply:
This is not a public analyzer at this time. We can revisit this in the future if/when we decide to make it a public analyzer.